### PR TITLE
Forward-merge release-0.17 into master

### PR DESCRIPTION
**Pull request by beorn7**
@fabxc I think this is what we want.

### CODE REVIEW

**Comment by fabxc** [comment]
:+1:

**Comment by beorn7** [comment]
CI errors are in master already, have to be fixed separately.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,57 @@
+## 0.17.0 / 2016-01-27
+
+This version no longer works with Alertmanager 0.4.0 and earlier!
```
Comment (grobie) [inline CHANGELOG.md]: Still a typo as reported a few times already.

**Comment by grobie** [comment]
Why was the changelog merged into master, describing 0.17.0 as being release with a release date in the past?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,57 @@
...
+The alerting rule syntax has changed as well but the old syntax is supported
+up until version 0.18.
+
+* [CHANGE] Integrate with Alertmanager 0.1.0 and higher
+* [CHANGE] Degraded storage mode renamed to rushed mode
+* [CHANGE] New alerting rule syntax
+* [CHANGE] Add label validation on ingestion
+* [CHANGE] Regular expression matchers in PromQL are anchored
```
Comment (grobie) [inline CHANGELOG.md]: This required changes to tens of rules and alert expressions and broke an unknown number of dashboard expressions. I'd recommend to explain the implications with an example in more detail above.

**Comment by beorn7** [comment]
For context: This PR simply merges forward into master what was done in the 0.17 release branch. It has no opinion about its correctness. If you want to change something that should affect 0.17, please change it there (will be merged forward later). If you want to change something that should only affect 0.18 and later, do it in master. Commenting on this PR about things already committed to the 0.17 release branch are out of place.
Thank you.

**Pull request merged**